op.c

```diff
@@ -3775,6 +3775,22 @@ S_optimize_op(pTHX_ OP* o)
             }
             break;
 
+        case OP_RV2AV:
+        {
+            OP *first = (o->op_flags & OPf_KIDS) ? cUNOPo->op_first : NULL;
+            if(CvSIGNATURE(PL_compcv) &&
+                    first && first->op_type == OP_GV &&
+                    cGVOPx_gv(first) == PL_defgv) {
+                OP *parent = op_parent(o);
+                while(parent && parent->op_type == OP_NULL)
```

Comment:
maybe consider using the existing macro for this 
`while( OP_TYPE_IS(parent, OP_NULL) )`

Reply:
Ahyes, a handy macro. Will do.